tests/h_item/h_item_api.cpp

```diff
@@ -13,293 +13,396 @@
 namespace TEST_NAMESPACE {
 using namespace sycl_cts;
 
-template <int numDims>
+template <int dims>
 struct kernel_common;
 
-/**
- * @brief Kernel struct that tests retrieving a single value from a range or ID
- *        for different ranges and IDs obtained from an h_item
- * @tparam numDims Number of dimensions of the group used in the kernel
- * @tparam currentDim Current dimension to get the value for
+/** @brief Storage for linear ID values
  */
-template <int numDims, int currentDim>
-struct kernel_single {
-  using success_acc_t =
-      cl::sycl::accessor<bool, 1, cl::sycl::access::mode::write>;
-
-  cl::sycl::range<numDims> kernelLogicalLocalRange;
-  success_acc_t success_acc;
-
-  void operator()(cl::sycl::group<numDims> group) const {
-    group.parallel_for_work_item(
-        kernelLogicalLocalRange, [&](cl::sycl::h_item<numDims> item) {
-          bool success = true;
-          {
-            auto value = item.get_global_range(currentDim);
-            auto expected = item.get_global_range()[currentDim];
-            success &= value == expected;
-          }
-          {
-            auto value = item.get_global_id(currentDim);
-            auto expected = item.get_global_id()[currentDim];
-            success &= value == expected;
-          }
-          {
-            auto value = item.get_local_range(currentDim);
-            auto expected = item.get_local_range()[currentDim];
-            success &= value == expected;
-          }
-          {
-            auto value = item.get_local_id(currentDim);
-            auto expected = item.get_local_id()[currentDim];
-            success &= value == expected;
-          }
-          {
-            auto value = item.get_logical_local_range(currentDim);
-            auto expected = item.get_logical_local_range()[currentDim];
-            success &= value == expected;
-          }
-          {
-            auto value = item.get_logical_local_id(currentDim);
-            auto expected = item.get_logical_local_id()[currentDim];
-            success &= value == expected;
-          }
-          {
-            auto value = item.get_physical_local_range(currentDim);
-            auto expected = item.get_physical_local_range()[currentDim];
-            success &= value == expected;
-          }
-          {
-            auto value = item.get_physical_local_id(currentDim);
-            auto expected = item.get_physical_local_id()[currentDim];
-            success &= value == expected;
-          }
-
-          // Each work-item updates success flag only if it is needed to avoid
-          // data race
-          if (!success) success_acc[0] = false;
-        });
+struct work_item_ids {
+  size_t global;
+  size_t physical;
+  size_t logical;
+};
+
+/** @brief Wrapper for global ID support in generic code
+ */
+struct global_id {
+  static const auto& value(const work_item_ids& ids) { return ids.global; }
+  static std::string description() { return "global ID"; };
```

Comment:
That' a great advice in general, thanks)
Still we do use lazy initialization paradigm in this test, therefore there would be no performance gain from such change. Meanwhile such change would require explicit cast to the `std::string` each time we get the `description`. Therefore I would like to avoid `string_view` usage in this specific case